Dockerfile

```diff
@@ -1,37 +1,56 @@
 FROM osrf/ros:kinetic-desktop
 
-LABEL com.nvidia.volumes.needed="nvidia_driver"
-ENV PATH /usr/local/nvidia/bin:${PATH}
-ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64:${LD_LIBRARY_PATH}
-
-RUN apt-get update \
- && apt-get install -y \
-    wget \
-    lsb-release \
-    sudo \
-    mesa-utils \
- && apt-get clean
-
-
-# Get gazebo binaries
-RUN echo "deb http://packages.osrfoundation.org/gazebo/ubuntu `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list \
- && wget http://packages.osrfoundation.org/gazebo.key -O - | apt-key add - \
- && apt-get update \
- && apt-get install -y \
-    gazebo8 \
-    ros-kinetic-gazebo8-ros-pkgs \
-    ros-kinetic-fake-localization \
-    ros-kinetic-joy \
- && apt-get clean
-
-
-RUN mkdir -p /tmp/workspace/src
-COPY prius_description /tmp/workspace/src/prius_description
-COPY prius_msgs /tmp/workspace/src/prius_msgs
-COPY car_demo /tmp/workspace/src/car_demo
-RUN /bin/bash -c 'cd /tmp/workspace \
- && source /opt/ros/kinetic/setup.bash \
- && catkin_make'
-
-
-CMD ["/bin/bash", "-c", "source /opt/ros/kinetic/setup.bash && source /tmp/workspace/devel/setup.bash && roslaunch car_demo demo.launch"]
+# setup keys
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys D2486D2DD83DB69272AFE98867170598AF249743
+
+# setup sources.list
+RUN . /etc/os-release \
+    && echo "deb http://packages.osrfoundation.org/gazebo/$ID-stable `lsb_release -sc` main" > /etc/apt/sources.list.d/gazebo-latest.list
+
+# setup catkin workspace
+ENV CATKIN_WS=/root/catkin_ws
+RUN mkdir -p $CATKIN_WS/src
+WORKDIR $CATKIN_WS/src
+
+# copy source code
+COPY prius_description .
+COPY prius_msgs .
+COPY car_demo .
+
+# install package dependacies
+RUN apt-get -qq update && \
+    apt-get -qq install -y \
+        python-catkin-tools && \
+    rosdep update && \
+    rosdep install -y \
+      --from-paths . \
+      --ignore-src \
+      --rosdistro ${ROS_DISTRO} \
+      --skip-keys=prius_msgs \
```

Comment:
I'm not sure why `--ignore-src` doesn't resolve this, but rosdep complaints without these exceptions.

Reply:
Is it in your workspace? It's only finding one package in the build.

Reply:
That would also explain the compile error too.

Reply:
But the `prius_msgs` package was copied into the source workspace directory in line 17: https://github.com/ruffsl/car_demo/blob/4449922ec4a23218d08a7626fe866d1cf3fcadff/Dockerfile#L17